multi_company_remove/models/res_company.py

```diff
@@ -85,7 +85,7 @@
         ' (select id from account_bank_statement where company_id = %s)',
     'res_partner':
         'delete from res_partner where company_id = %s and id not in'
-        ' (select partner_id from res_company)',
+        ' (select partner_id from res_company) and user_id IS NULL',
```

Comment:
You will need to change this. The user_id specified here is the sales person. So instead of simply testing for user_id, you will have to look into the users table and delete the partners that do not refer to res_partner with partner_id.

Reply:
'delete from res_partner where company_id = %s and id not in'
        ' (select partner_id from res_company) and res_partner.id not in'
        ' (select id from res_users where partner_id is null)',

Reply:
After each 'and' start the next condition on a new line.
The additional condition should be:
```
id not in (select partner_id from res_users)
```